source/geometries/Next100FieldCage.cc

```diff
@@ -613,6 +656,149 @@ void Next100FieldCage::BuildFieldCage()
   }
 }
 
+void Next100FieldCage::BuildFieldCage()//////////////////////////////////////////
+{
+  // HDPE cilinder.
+  G4double hdpe_tube_z_pos = -hdpe_length_/2. + teflon_total_length_ + gate_teflon_dist_ + GetELzCoord();
+
+  G4Tubs* hdpe_tube_solid =
+    new G4Tubs("HDPE_TUBE", hdpe_tube_int_diam_/2.,
+               hdpe_tube_ext_diam_/2., hdpe_length_/2., 0, twopi);
+
+  G4LogicalVolume* hdpe_tube_logic = new G4LogicalVolume(hdpe_tube_solid,
+                                                         hdpe_, "HDPE_TUBE");
+
+  new G4PVPlacement(0, G4ThreeVector(0., 0., hdpe_tube_z_pos), hdpe_tube_logic,
+                                     "HDPE_TUBE", mother_logic_, false, 0, true);
+
+  G4double active_short_z = 13.5*mm; //Thickness for the small piece from the  holder in front of the ring.
+  G4double first_ring_drif_z_pos = drift_ring_dist_/2. + active_short_z/2. + gate_teflon_dist_ + GetELzCoord();
+  G4double buffer_short_z = 37.*mm;
+
+  G4double teflon_drift_length = active_cathode_dist_;
+  G4double teflon_drift_zpos = GetELzCoord() + gate_teflon_dist_ + teflon_drift_length/2.;
+  G4double teflon_buffer_length = teflon_total_length_ - cathode_thickn_ - teflon_drift_length;
+  G4double teflon_buffer_zpos = teflon_drift_zpos + teflon_drift_length/2. + cathode_thickn_ + teflon_buffer_length/2;
```

Comment:
Should I define a function as `DefineMaterials()` in` NextNewFieldCage` for all the materials used in the class?

Reply:
No, they are not material variables, they are distances. These variables `teflon_*`, defined in lines 678-681 are defined inside the function `BuildFieldCage`, but they are also defined in the previous function `BuildLightTube` in lines 538, 539, 574, 575. This is very bug prone since if you change one on these distances you must do it on two different places. The solution to avoid this duplication is to define them as global variables and define them once, as it is done for variables in lines 200-207 in the `Construct`. I suggest to place the `teflon_*` variables there also. Remember that global variable names finish with an underscore, like `active_length_`.

Reply:
Done!